### ISSUE TITLE
sidebar - onclick - selected

### ISSUE THREAD

**Issue by CThuleHansen** [vitmalina/w2ui#1267]
Hi,

I am interested in achieving the "selected"-behavior of the click method (https://github.com/vitmalina/w2ui/blob/master/src/w2sidebar.js#L449) in the dblClick method.
Currently I have commented some lines in the source, but I would like the framework to support this out of the box, so I can update it without manual editing.
The reason why it is not supported is because the before trigger in the click method
`var edata = obj.trigger({ phase: 'before', type: 'click', target: id, originalEvent: event, node: nd, object: nd });`
is invoked after removing and adding of the w2ui-selected css class
`// unselect all previsously
            $(obj.box).find('.w2ui-node.w2ui-selected').each(function (index, el) {
                var oldID     = $(el).attr('id').replace('node_', '');
                var oldNode = obj.get(oldID);
                if (oldNode != null) oldNode.selected = false;
                $(el).removeClass('w2ui-selected').find('.w2ui-icon').removeClass('w2ui-icon-selected');
            });
            // select new one
            var newNode = $(obj.box).find('#node_'+ w2utils.escapeId(id));
            var oldNode = $(obj.box).find('#node_'+ w2utils.escapeId(obj.selected));
            newNode.addClass('w2ui-selected').find('.w2ui-icon').addClass('w2ui-icon-selected');`

This does 2 things: 

- It creates a "flickering" effect of unselected/selected. 
- it interfers with the dbl click, as the process when dbl clicking happens like this: click, dblclick, click. 

Is it possible to move the before trigger above the CSS selection/unselection?

**Comment by CThuleHansen** [comment]
Any feedback on this?

**Comment by mrg2001** [comment]
Let me rephrase what you are asking. You want double click to select sidebar item and not click, right?

**Comment by CThuleHansen** [comment]
@mrg2001 exactly.  Use case: 
When users double click a sidebar item, the item is selected and a view related to that side bar item is opened.
